File: src/test/java/scrabble/phrases/words/WordUtilsTest.java
package scrabble.phrases.words;

import static org.junit.jupiter.api.Assertions.assertEquals;

import java.util.HashMap;
import java.util.Map;

import org.junit.jupiter.api.Test;

/**
 * Tests for WordUtils.
 */
class WordUtilsTest {

    @Test
    void shouldComputeCorrectSyllableCount() {
        Map<String, Integer> wordMap = createWordMap(
            "semifinalista 6", "greoi 2", "aalenian 4", "alee 3", "alcool 3", "fiinta 3",
            "puicuta 3", "aeroport 4", "miercureana 4", "policioara 4", "vulpoaica 3", "miau 1", "leoaica 3",
            "lupoaica 3", "mioara 2", "ambiguul 4", "tămâie 3", "bou 1", "reusit 3", "greul 2", "plouat 2",
            "roua 2", "calea 2", "eu 1", "greu 1", "pui 1", "tuiul 2", "ghioc 2"
        );

        for (var entry : wordMap.entrySet()) {
            assertEquals(entry.getValue().intValue(), WordUtils.computeSyllableNumber(entry.getKey()),
                "Syllable count for word: " + entry.getKey());
        }
    }

    @Test
    void shouldCapitalizeFirstLetter() {
        assertEquals("Aloha", WordUtils.capitalizeFirstLeter("aloha"));
        assertEquals("Aloha", WordUtils.capitalizeFirstLeter("Aloha"));
        assertEquals("Înăbușeala", WordUtils.capitalizeFirstLeter("înăbușeala"));
    }

    @Test
    void shouldComputeRhyme() {
        assertEquals("are", WordUtils.computeRhyme("formare"));
        assertEquals("abc", WordUtils.computeRhyme("abc"));
        assertEquals("ab", WordUtils.computeRhyme("ab"));
    }

    private Map<String, Integer> createWordMap(String... entries) {
        Map<String, Integer> wordMap = new HashMap<>();
        for (String entry : entries) {
            String[] parts = entry.split(" ");
            wordMap.put(parts[0], Integer.parseInt(parts[1]));
        }
        return wordMap;
    }
}

File: src/test/kotlin/scrabble/phrases/words/WordUtilsTest.kt
package scrabble.phrases.words

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

class WordUtilsTest {

    @Test
    fun shouldComputeCorrectSyllableCount() {
        val wordMap = mapOf(
            "semifinalista" to 6, "greoi" to 2, "aalenian" to 4, "alee" to 3, "alcool" to 3,
            "fiinta" to 3, "puicuta" to 3, "aeroport" to 4, "miercureana" to 4, "policioara" to 4,
            "vulpoaica" to 3, "miau" to 1, "leoaica" to 3, "lupoaica" to 3, "mioara" to 2,
            "ambiguul" to 4, "t\u0103m\u00e2ie" to 3, "bou" to 1, "reusit" to 3, "greul" to 2,
            "plouat" to 2, "roua" to 2, "calea" to 2, "eu" to 1, "greu" to 1, "pui" to 1,
            "tuiul" to 2, "ghioc" to 2,
            "laur" to 2, "taur" to 2, "dinozaur" to 4
        )

        for ((word, expected) in wordMap) {
            assertEquals(expected, WordUtils.computeSyllableNumber(word),
                "Syllable count for word: $word")
        }
    }

    @Test
    fun shouldCapitalizeFirstLetter() {
        assertEquals("Aloha", WordUtils.capitalizeFirstLetter("aloha"))
        assertEquals("Aloha", WordUtils.capitalizeFirstLetter("Aloha"))
        assertEquals("\u00cen\u0103bu\u0219eala", WordUtils.capitalizeFirstLetter("\u00een\u0103bu\u0219eala"))
    }

    @Test
    fun shouldComputeRhyme() {
        assertEquals("are", WordUtils.computeRhyme("formare"))
        assertEquals("abc", WordUtils.computeRhyme("abc"))
        assertEquals("ab", WordUtils.computeRhyme("ab"))
    }
}
